src/test/glob.test.ts

```diff
@@ -120,491 +155,584 @@ test.after.each(async (ctx) => {
-    throwIfOutsideCwd: false,
+for (const mode of ['once', 'watch'] as const) {
+  test(`[${mode}] empty patterns`, ({check}) =>
+    check({mode, files: ['foo'], patterns: [], expected: []}));
+
+  if (mode !== 'watch') {
+    test(`[${mode}] normalizes trailing / in pattern`, ({check}) =>
+      check({mode, files: ['foo'], patterns: ['foo/'], expected: ['foo']}));
+  }
+
+  test(`[${mode}] normalizes ../ in pattern`, ({check}) =>
+    check({mode, files: ['foo'], patterns: ['bar/../foo'], expected: ['foo']}));
+
+  test(`[${mode}] explicit file that does not exist`, ({check}) =>
+    check({mode, files: [], patterns: ['foo'], expected: []}));
+
+  test(`[${mode}] * star`, ({check}) =>
+    check({
+      mode,
+      files: ['foo', 'bar'],
+      patterns: ['*'],
+      expected: ['foo', 'bar'],
+    }));
+
+  test(`[${mode}] * star with ! negation`, ({check}) =>
+    check({
+      mode,
+      files: ['foo', 'bar', 'baz'],
+      patterns: ['*', '!bar'],
+      expected: ['foo', 'baz'],
+    }));
+
+  test(`[${mode}] inclusion of directory with trailing slash`, ({check}) =>
+    check({
+      mode,
+      files: ['foo/good/1', 'foo/good/2'],
+      patterns: ['foo/'],
+      expected: ['foo/good/1', 'foo/good/2'],
+      expandDirectories: true,
+    }));
+
+  test(`[${mode}] inclusion of directory without trailing slash`, ({check}) =>
+    check({
+      mode,
+      files: ['foo/good/1', 'foo/good/2'],
+      patterns: ['foo'],
+      expected: ['foo/good/1', 'foo/good/2'],
+      expandDirectories: true,
+    }));
+
+  test(`[${mode}] !exclusion of directory with trailing slash`, ({check}) =>
+    check({
+      mode,
+      files: ['foo/good/1', 'foo/bad/1'],
+      patterns: ['foo', '!foo/bad/'],
+      expected: ['foo/good/1'],
+      expandDirectories: true,
+    }));
+
+  test(`[${mode}] !exclusion of directory without trailing slash`, ({check}) =>
+    check({
+      mode,
+      files: ['foo/good/1', 'foo/bad/1'],
+      patterns: ['foo', '!foo/bad'],
+      expected: ['foo/good/1'],
+      expandDirectories: true,
+    }));
+
+  test(`[${mode}] explicit .dotfile`, ({check}) =>
+    check({mode, files: ['.foo'], patterns: ['.foo'], expected: ['.foo']}));
+
+  test(`[${mode}] * star matches .dotfiles`, ({check}) =>
+    check({mode, files: ['.foo'], patterns: ['*'], expected: ['.foo']}));
+
+  test(`[${mode}] {} groups`, ({check}) =>
+    check({
+      mode,
+      files: ['foo', 'bar', 'baz'],
+      patterns: ['{foo,baz}'],
+      expected: ['foo', 'baz'],
+    }));
+
+  test(`[${mode}] matches explicit symlink`, ({check}) =>
+    check({
+      mode,
+      files: [
+        'target',
+        {target: 'target', path: 'symlink', windowsType: 'file'},
+      ],
+      patterns: ['symlink'],
+      expected: ['symlink'],
+    }));
+
+  test(`[${mode}] explicit directory excluded when includeDirectories=false`, ({
+    check,
+  }) => check({mode, files: ['foo/'], patterns: ['foo'], expected: []}));
+
+  if (mode !== 'watch') {
```

Comment:
Nit: Could abstract this check to be more readable. E.g. `skipWatchTest()', which could have a comment.
Optional.

Reply:
Done!